### ISSUE TITLE
Use Protected not Private

### ISSUE THREAD

**Issue by demckee** [nopSolutions/nopCommerce#3399]
nopCommerce version: 4.10

The bulk of the classes in nop are designed to be overridden yet they all have private fields. It's bad enough when you have to override a class and build a new constructor with a ton of parameters (as required by DI). You then have to declare your own copy of those parameters for values you need because the base class has them defined as private not protected. This is just plain bloat. Why make a method virtual then make people jump through extra work to reference fields that the base class has as private? 

Nothing in nop should be marked as private. If you want to type in private, use protected instead. 

For example, PageHeadBuilder doesn't work. I want to override GenerateScripts, GenerateCssFiles and GetBundleFileName. However that means I also need to override a bunch of other perfectly acceptable methods because _cssFiles and _scriptParts are both private.

**Comment by noptools** [comment]
I can understand changing from private field to protected field for "core" classes, but would you please 'justify' why you also did the update for Plugins?  Are devs actually are inheriting from plugin classes?

**Comment by AndreiMaz** [comment]
@noptools No. But we've done it just to make it consistent with the "core" classes
